crates/typed-store/src/rocks/mod.rs

```diff
@@ -371,6 +372,20 @@ impl RocksDB {
         delegate_call!(self.flush())
     }
 
+    pub fn checkpoint(&self, path: &Path) -> Result<(), rocksdb::Error> {
```

Comment:
how do we get a handle to the snapshot later? just open it under the given path?

there are two ways we might use this:
1. to support private backups / restores for node operators
2. fullnodes might take a rocksdb snapshot at reconfiguration time, and then publish a sui snapshot from that database in the background while continuing into the next epoch. (Without this, the node would have to pause for a long time at reconfig).

Reply:
> how do we get a handle to the snapshot later? just open it under the given path?

Yes, that is exactly it.

> to support private backups / restores for node operators

This is a great point. I've been thinking about this lately. But there is still another complication here which I've been meaning to talk to you about. We have many different databases i.e. perpetual, epoch, checkpoint, etc and taking a checkpoint of different ones cannot be done consistently (point in time across all of them). 
1. We can probably build a recovery mechanism which can do a safe recovery to a point in time which is consistent across different databases. 
2. We can enable rocksdb timestamps for all dbs (another feature which we don't use yet in rocksdb)

> fullnodes might take a rocksdb snapshot at reconfiguration time, and then publish a sui snapshot from that database in the background while continuing into the next epoch. (Without this, the node would have to pause for a long time at reconfig).

This is exactly the plan with this change. The state snapshot will be take in the background without blocking execution from full nodes (or any node for that matter which wants to take the snapshot)